metrics_api/test/opentelemetry/metrics/meter_test.rb

```diff
@@ -7,15 +7,15 @@
 require 'test_helper'
 
 describe OpenTelemetry::Metrics::Meter do
-  INSTRUMENT_NAME_ERROR = OpenTelemetry::Metrics::Meter::InstrumentNameError
-  INSTRUMENT_UNIT_ERROR = OpenTelemetry::Metrics::Meter::InstrumentUnitError
-  INSTRUMENT_DESCRIPTION_ERROR = OpenTelemetry::Metrics::Meter::InstrumentDescriptionError
-  DUPLICATE_INSTRUMENT_ERROR = OpenTelemetry::Metrics::Meter::DuplicateInstrumentError
+  let(:instrument_name_error)        { OpenTelemetry::Metrics::Meter::InstrumentNameError }
+  let(:instrument_unit_error)        { OpenTelemetry::Metrics::Meter::InstrumentUnitError }
+  let(:instrument_description_error) { OpenTelemetry::Metrics::Meter::InstrumentDescriptionError }
+  let(:duplicate_instrument_error)   { OpenTelemetry::Metrics::Meter::DuplicateInstrumentError }
```

Comment:
Constants defined in a `describe` block are global to the whole test suite which may not be the desirable effect, so I've replaced them with a `let` definition.

Reply:
After reading [Synchronous Instrument API](https://opentelemetry.io/docs/reference/specification/metrics/api/#synchronous-instrument-api) I noticed the following

> The API SHOULD be documented in a way to communicate to users that the
> `name` parameter needs to conform to the instrument name syntax.
> **The API SHOULD NOT validate the `name`; that is left to
> implementations of the API.**

and

> The `unit` parameter needs to support the instrument unit rule.
> Meaning, the API MUST accept a case-sensitive string that supports
> ASCII character encoding and can hold at least 63 characters.
> **The API SHOULD NOT validate the `unit`.**

Similar for `name` and `unit` in [Asynchronous Instrument API](https://opentelemetry.io/docs/reference/specification/metrics/api/#asynchronous-instrument-api)

So I'm thinking if we should actually remove all these validations from API.

Reply:
Because of this, I'm thinking of closing this PR and replacing it with https://github.com/open-telemetry/opentelemetry-ruby/pull/1442